File: Python/Data Structure/Tree/tree.py
# node class
class Node:

    def __init__(self, data):
        # left child
        self.left = None
        # right child
        self.right = None
        # node's value
        self.data = data

    # print function
    def PrintTree(self):
        print(self.data)

root = Node(27)

root.PrintTree()


class Node:

    def __init__(self, data):

        self.left = None
        self.right = None
        self.data = data

    def insert(self, data):
# Compare the new value with the parent node
        if self.data:
            if data < self.data:
                if self.left is None:
                    self.left = Node(data)
                else:
                    self.left.insert(data)
            elif data > self.data:
                if self.right is None:
                    self.right = Node(data)
                else:
                    self.right.insert(data)
        else:
            self.data = data

# Print the tree
    def PrintTree(self):
        if self.left:
            self.left.PrintTree()
        print( self.data),
        if self.right:
            self.right.PrintTree()

# Use the insert method to add nodes
root = Node(27)
root.insert(14)
root.insert(35)
root.insert(31)
root.insert(10)
root.insert(19)


root.PrintTree()




class Node:

    def __init__(self, data):

        self.left = None
        self.right = None
        self.data = data

# Insert method to create nodes
    def insert(self, data):

        if self.data:
            if data < self.data:
                if self.left is None:
                    self.left = Node(data)
                else:
                    self.left.insert(data)
            elif data > self.data:
                if self.right is None:
                    self.right = Node(data)
                else:
                    self.right.insert(data)
        else:
            self.data = data
# findval method to compare the value with nodes

    def findval(self, lkpval):
        if lkpval < self.data:
            if self.left is None:
                return str(lkpval)+" is not Found"
            return self.left.findval(lkpval)
        elif lkpval > self.data:
            if self.right is None:
                return str(lkpval)+" is not Found"
            return self.right.findval(lkpval)
        else:
            return str(self.data) + " is found"
# Print the tree

    def PrintTree(self):
        if self.left:
            self.left.PrintTree()
        print(self.data),
        if self.right:
            self.right.PrintTree()

root = Node(27)
root.insert(14)
root.insert(35)
root.insert(31)
root.insert(10)
root.insert(19)
print(root.findval(7))
print(root.findval(14))

File: C++/Data Structure/TREE/tree.cpp
#include <iostream>
using namespace std;

class TreeNode{
   public:
    TreeNode* data;
    TreeNode* left;
    TreeNode* right;
    TreeNode * root;

   };
 TreeNode* BST:: insert(Node*node , int val){
        if(node==NULL){
         node = new Tree;
         node->data = val;
         node->left = NULL;
         node->right = NULL;
         node->parent= NULL;
        }

    }
void Tree::Inorder(Node* root){
    cout<<"this is going to print tree Node in inorder"<<endl;
 }
int main(){
   TreeNode* root = NULL;
   root = insert(root , 1);
   root = insert(root , 2);
   root = insert(root , 3);

 return 0;
}
/*
class Tree{
   public:
       Tree(){
        Node* root=NULL;
       }
    void insert

   };

void Tree::Inorder(Node* root){
    cout<<"this is going to print tree Node in inorder"<<endl;
 }


 */
